scripts/validate_docstrings.py

```diff
@@ -407,6 +411,26 @@ def examples_source_code(self):
         lines = doctest.DocTestParser().get_examples(self.raw_doc)
         return [line.source for line in lines]
 
+    def validate_pep8(self):
+        if not self.examples:
+            return
+
+        content = ''.join(('import numpy as np  # noqa: F401\n',
```

Comment:
may be we can add a comment before this line explaining why the `F401` is required.

Reply:
Well the content basically never shows as it's only existing within tmp files. And if someone thinks it's unnecessary the tests would fail. 
So I don't think commenting is needed there.

Reply:
The comment was not for the temp file, but our code, and I think it's useful to know what the `F401` is about when reading this part of code (without having to remove it and see that fails). Anyway, this is merged now. :)